### PR TITLE
Prevent metric calculation for empty slices by checking if array is ~0

### LINKED ISSUE #3402: Morphometric values are reported in slices that are not segmented
### Description

When using `sct_process_segmentation` on the segmentation attached below, the metrics are reported in the output CSV file for slices where no segmentation is present (ie: the slice is empty). 

Image of the segmentation showing that slice 3 is empty (ie: no segmentation):
Screen Shot 2021-05-26 at 9 59 46 AM

And the output CSV, showing a morphometric value at slice 3:

Slice (I-&gt;S) | VertLevel | MEAN(area) | STD(area) | MEAN(angle_AP)
-- | -- | -- | -- | --
0 | &nbsp; | None | None | None
1 | &nbsp; | None | None | None
2 | &nbsp; | None | None | None
3 | &nbsp; | 45.49999186396620 | 0.0 | 0.0
4 | &nbsp; | 45.49999186396620 | 0.0 | 0.0
5 | &nbsp; | 55.81249001994760 | 0.0 | 0.0
6 | &nbsp; | 60.499989181757300 | 0.0 | 0.0

Interestingly, values for slices 3 and 4 are the same, which suggests a bug.

### Steps to Reproduce

0. Install SCT 77d4b7ddc716e69b1dc223536a7c026ff12abad8 (bug should appear with recent versions)

1. Download this data: [simple_avg_medic_oben_P06_month00_seg_corr.nii.gz](https://github.com/spinalcordtoolbox/spinalcordtoolbox/files/6547252/simple_avg_medic_oben_P06_month00_seg_corr.nii.gz)

2. Run:
   ```console
   sct_process_segmentation -i simple_avg_medic_oben_P06_month00_seg_corr.nii.gz -perslice 1 -angle-corr 0 -o csa_seg_gz_anglecorr0.csv 
   ```

**Expected behavior:** 
`MEAN(area)=None` at row=3

**Actual behavior:** <What actually happens>
`MEAN(area)=45.49999186396620`

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [X] I've given this PR a concise, self-descriptive, and meaningful title
- [X] I've linked relevant issues in the PR body
- [X] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [X] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [X] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

Instead of checking if image is exactly 0, use a threshold, as resampling can change 0 -> ~0 (e.g. 1e-16).

## Linked issues

Fixes #3402.

### CODE REVIEW

File: spinalcordtoolbox/process_seg.py [hunk 1]
```diff
@@ -15,6 +15,10 @@
 from spinalcordtoolbox.resampling import resample_nib
 from spinalcordtoolbox.utils import sct_progress_bar
 
+# NB: We use a threshold to check if an array is empty, instead of checking if it's exactly 0. This is because
+# resampling can change 0 -> ~0 (e.g. 1e-16). See: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3402
+NEAR_ZERO_THRESHOLD = 1e-6
```
Comment (jcohenadad) [inline spinalcordtoolbox/process_seg.py:20]: ah! i knew it 😉

Comment (jcohenadad) [reply spinalcordtoolbox/process_seg.py:20]: good debugging skills 👏

**Review by jcohenadad — APPROVED** [review]
i assume you tested it @joshuacwnewton so no need to test it on my end-- thank you for finding this tricky 🐛

**Comment by joshuacwnewton** [comment]
> i assume you tested it

Oops, I should have been explicit about this. Yes, I did: [csa_seg_gz_anglecorr0.csv](https://github.com/spinalcordtoolbox/spinalcordtoolbox/files/6549210/csa_seg_gz_anglecorr0.csv)

Screenshot from 2021-05-26 15-30-35

**Pull request merged**